### PR TITLE
Groups added

### PR DESCRIPTION
**Pull request by Lucien950**

### CODE REVIEW

**Comment by Lucien950** [comment]
also fixed some redirect related things

File: middleware/router-auth.ts [hunk 1]
```diff
@@ -5,29 +5,46 @@ export default function ({ store, redirect, route }: Context) {
     // if(route.name == null) redirect ('/')
 
     const validUser = (store.state.auth.user != null && store.state.auth.user.emailVerified)
-    console.log(route.name);
+    console.log('\n')
+    console.log("Going to " + route.name + (""?validUser:" not") + " as a valid user");
+
+    // makes main page splash or home depending on if you are valid
     if (route.name == "index") {
+        console.log("case 1")
```
Comment (picklepower24) [inline middleware/router-auth.ts:13]: maybe don't have these cryptic case messages

File: middleware/router-auth.ts [hunk 2]
```diff
@@ -5,29 +5,46 @@ export default function ({ store, redirect, route }: Context) {
...
         return !validUser ? redirect("/splash") : redirect("/home");
     }
-
+    
+    // go home if you are already valid and you are being sent to auth
     if ((validUser) && (route.name == null || route?.name?.split('/').some(record => record == 'auth'))) {
+        console.log("case 2")
```
Comment (picklepower24) [inline middleware/router-auth.ts:19]: same

File: middleware/router-auth.ts [hunk 3]
```diff
@@ -5,29 +5,46 @@ export default function ({ store, redirect, route }: Context) {
...
         return redirect('/home')
     }
+
+    // makes invalid users either verify email(if haven't) or force login if path is inside app
     if(!validUser && (route.name == null || isAdminRoute(route)))
     {
         if(store.state.auth.user && !store.state.auth.user.emailVerified)
         {
+            console.log("case 3a")
```
Comment (picklepower24) [inline middleware/router-auth.ts:28]: sam e

File: middleware/router-auth.ts [hunk 4]
```diff
@@ -5,29 +5,46 @@ export default function ({ store, redirect, route }: Context) {
...
             return redirect("/auth/verify-email")
         }
+        console.log("case 3b")
```
Comment (picklepower24) [inline middleware/router-auth.ts:31]: same

File: middleware/router-auth.ts [hunk 5]
```diff
@@ -5,29 +5,46 @@ export default function ({ store, redirect, route }: Context) {
...
         const redirectPath = route.fullPath;
         // Records the proper link
         return redirect(`/auth/login?redirect=${encodeURIComponent(redirectPath)}`)
     };
+    console.log('\n')
```
Comment (picklepower24) [inline middleware/router-auth.ts:36]: ??

File: pages/groups/index.vue [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+import { Vue, Component } from 'nuxt-property-decorator'
+import { groupsStore } from '~/store'
+import NewsletterModal from '~/screens/newsletter/NewsletterModal.vue'
+import NewsletterCard from '~/components/NewsletterCard.vue'
+import GroupCard from '~/components/GroupCard.vue'
+
+@Component<GroupsPage>({
+  components: { GroupCard },
+  mounted() {
+    console.log('abanana')
```
Comment (picklepower24) [inline pages/groups/index.vue:39]: remove

File: store/groups.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    this.groupList = items;
+  }
+
+  @Action({ rawError: true })
+  public async GetGroups() {
+    try {
+      const query = await firestore.collection('groups').orderBy(this.sortBy).get();
+      const items = query.docs.map(doc => Group.fromFirebase(doc.data() as Group_t_F, doc.id));
+      this.SET_GROUP_LIST(items);
+      console.log('banana');
```
Comment (picklepower24) [inline store/groups.ts:36]: remove

**Review by picklepower24 — CHANGES_REQUESTED** [review]
The page is working! However, there doesn't seem to be a member list, notes page, or questions..
image
On click, I'm just brought to home page.
Also, remove extra console logs.

File: pages/groups/index.vue (hunk 6 above)
Comment (Lucien950) [reply pages/groups/index.vue:46]: done

File: store/groups.ts (hunk 7 above)
Comment (Lucien950) [reply store/groups.ts:36]: done

File: pages/groups/index.vue (hunk 6 above)
Comment (Lucien950) [reply pages/groups/index.vue:46]: wait no it breaks the things if remove it

**Comment by Lucien950** [comment]
# Major Issues

- [ ] - Not all groups display in the cards ie. Discover.vue page
- [ ] - When creating a group, there is no option to choose a backgroundImage
- [ ] - Members (the number) does not update correctly
- [ ] - Author information not included when creating a group

**Pull request merged**